libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -319,11 +325,15 @@ static void establishMqttSession( MQTTContext_t * pContext,
      * function will be used to calculate intervals in MQTT library.*/
     callbacks.getTime = Clock_GetTimeMs;
 
-    /* Initialize MQTT library. */
-    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Init( pContext,
-                                               &transport,
-                                               &callbacks,
-                                               &networkBuffer ) );
+    /* Clear the state of the MQTT context when creating a clean session. */
+    if( createCleanSession == true )
```

Comment:
Can we always control the MQTT_Init logic based on createCleanSession?  How about adding a different flag to control the MQTT_Init?

Reply:
He initially had a separate flag, but I asked him to remove it since its value was always the same as createCleanSession. Since the only time we don't want to call MQTT_Init is when we don't want a clean session, I don't see why they should be separate.

Reply:
I think we won't be able to associate it always with a clean sesion flag. How about a client having QoS0 publishes only, but has active subscriptions? Reestablishing a session just means the subscriptions being continued in the broker. In this case reinitiailizing the context doesn't really matter. Just the client Id is what matters.

Reply:
If we want to test that situation later, we can always add it then. Right now, a separate parameter is unnecessary given our current tests.

Reply:
I don't totally agree with this change. I like it to be separate init logic and separate clean session flag for anyone not so familiar with the code to easily reuse the function for any new integration tests. However I am ok to proceed with the change that we may need to modify it with a different test. I am good to disagree and commit.